pod.go

```diff
@@ -0,0 +1,310 @@
+/*
+Copyright 2017 Devin All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Pod is the definition for an pod in marathon
+type Pod struct {
+	ID                string                  `json:"id,omitempty"`
+	Labels            map[string]string       `json:"labels,omitempty"`
+	Version           string                  `json:"version,omitempty"`
+	User              string                  `json:"user,omitempty"`
+	Environment       map[string]interface{}  `json:"environment,omitempty"`
+	Containers        []*PodContainer         `json:"containers,omitempty"`
+	Secrets           map[string]SecretSource `json:"secrets,omitempty"`
+	Volumes           []*PodVolume            `json:"volumes,omitempty"`
+	Networks          []*PodNetwork           `json:"networks,omitempty"`
+	Scaling           *PodScalingPolicy       `json:"scaling,omitempty"`
+	Scheduling        *PodSchedulingPolicy    `json:"scheduling,omitempty"`
+	ExecutorResources *ExecutorResources      `json:"executorResources,omitempty"`
+}
+
+// PodScalingPolicy is the scaling policy of the pod
+type PodScalingPolicy struct {
+	Kind         string `json:"kind"`
+	Instances    int    `json:"instances"`
+	MaxInstances int    `json:"maxInstances,omitempty"`
+}
+
+// NewPod create an empty pod
+func NewPod() *Pod {
+	return &Pod{
+		Labels:      map[string]string{},
+		Environment: map[string]interface{}{},
+		Containers:  []*PodContainer{},
+		Secrets:     map[string]SecretSource{},
+		Volumes:     []*PodVolume{},
+		Networks:    []*PodNetwork{},
+	}
+}
+
+// String marshals the pod as an indented string
+func (p *Pod) String() string {
+	s, err := json.MarshalIndent(p, "", "  ")
+	if err != nil {
+		return fmt.Sprintf(`{"error": "error decoding type into json: %s"}`, err)
+	}
```

Comment:
I've tended to use it for debugging, yes. This does match the implementation of the Application.String() method: https://github.com/gambol99/go-marathon/blob/master/application.go#L625

I also tend to like it as JSON because that's how I normally interact with these objects on the filesystem, and then I can dump them in the UI for more rapid testing.